### PR TITLE
Expose functionality as a re-usable library - Part 2

### PR DESCRIPTION
**Pull request by jiegec**
Part 2 of #279: add `lsof_context * ctx` argument to functions. Remove `_PROTOTYPE` macro to allow `clang-format` to recognize function prototypes.

### CODE REVIEW

**Review by BenBE — CHANGES_REQUESTED** [review]
Please re-check if all functions properly pass the context around. Several places seem to be missing the `ctx` argument.

**Review by BenBE — CHANGES_REQUESTED** [review]
Please re-check if all functions properly pass the context around. Several places seem to be missing the `ctx` argument.

File: lib/dialects/aix/dfile.c [hunk 1]
```diff
@@ -405,8 +402,7 @@ dev_t *dev;       /* pointer to device to be printed */
  * readvfs() - read vfs structure
  */
 
-struct l_vfs *readvfs(vn)
-struct vnode *vn; /* vnode */
+struct l_vfs *readvfs(struct vnode *vn) /* vnode */
```
Comment (BenBE) [inline lib/dialects/aix/dfile.c:405]: Missing `ctx` argument

File: lib/dialects/aix/dnode.c [hunk 2]
```diff
@@ -117,8 +117,7 @@ struct rnode {
  * isglocked() - is a gnode locked
  */
 
-char isglocked(ga)
-struct gnode *ga; /* local gnode address */
+char isglocked(struct gnode *ga) /* local gnode address */
```
Comment (BenBE) [inline lib/dialects/aix/dnode.c:120]: Missing `ctx` argument

File: lib/dialects/aix/dnode.c [hunk 3]
```diff
@@ -173,7 +172,7 @@ struct gnode *ga; /* local gnode address */
  * process_node() - process vnode
  */
 
-void process_node(va) KA_T va; /* vnode kernel space address */
+void process_node(KA_T va) /* vnode kernel space address */
```
Comment (BenBE) [inline lib/dialects/aix/dnode.c:175]: Missing `ctx` argument

File: lib/dialects/aix/dnode.c [hunk 4]
```diff
@@ -1135,8 +1135,8 @@ void process_node(va) KA_T va; /* vnode kernel space address */
  * process_shmt() -- process shared memory transport file
  */
 
-void process_shmt(sa) KA_T sa; /* shared memory transport node struct
-                                * address ??? */
+void process_shmt(KA_T sa) /* shared memory transport node struct
```
Comment (BenBE) [inline lib/dialects/aix/dnode.c:1138]: Missing `ctx` argument

File: lib/dialects/aix/dnode1.c [hunk 5]
```diff
@@ -78,9 +78,8 @@ struct vnode *alloc_vcache() {
  * getvolume() - get volume structure
  */
 
-static struct volume *getvolume(f, vols)
-struct VenusFid *f; /* file ID pointer */
-int *vols;          /* afs_volumes status return */
+static struct volume *getvolume(struct VenusFid *f, /* file ID pointer */
```
Comment (BenBE) [inline lib/dialects/aix/dnode1.c:81]: Missing `ctx` argument

File: lib/dialects/aix/dnode1.c [hunk 6]
```diff
@@ -123,8 +122,7 @@ int *vols;          /* afs_volumes status return */
  * hasAFS() - test for AFS presence via vfs structure
  */
 
-int hasAFS(vp)
-struct vnode *vp; /* vnode pointer */
+int hasAFS(struct vnode *vp) /* vnode pointer */
```
Comment (BenBE) [inline lib/dialects/aix/dnode1.c:125]: Missing `ctx` argument

File: lib/dialects/aix/dnode1.c [hunk 7]
```diff
@@ -162,9 +160,8 @@ struct vnode *vp; /* vnode pointer */
  *		  1 if root file ID structure address available
  */
 
-static int is_rootFid(vc, rfid)
-struct vcache *vc; /* vcache entry */
-int *rfid;         /* root file ID pointer status return */
+static int is_rootFid(struct vcache *vc, /* vcache entry */
```
Comment (BenBE) [inline lib/dialects/aix/dnode1.c:163]: Missing `ctx` argument

File: lib/dialects/aix/dnode1.c [hunk 8]
```diff
@@ -227,10 +224,9 @@ int *rfid;         /* root file ID pointer status return */
  * readafsnode() - read AFS node
  */
 
-int readafsnode(va, v, an)
-KA_T va;            /* kernel vnode address */
-struct vnode *v;    /* vnode buffer pointer */
-struct afsnode *an; /* afsnode recipient */
+int readafsnode(KA_T va,            /* kernel vnode address */
```
Comment (BenBE) [inline lib/dialects/aix/dnode1.c:227]: Missing `ctx` argument

File: lib/dialects/aix/dnode2.c [hunk 9]
```diff
@@ -44,25 +44,25 @@ static char copyright[] =
...
 #    include <j2/j2_inode.h>
 #    undef PROTO_H  /* enable "proto.h" */
 #    undef DPROTO_H /* enable "dproto.h" */
 #    include "proto.h"
 #    include "dproto.h"
 
-int readj2lino(ga, li)
-struct gnode *ga; /* gnode address */
-struct l_ino *li; /* local inode receiver */
+int readj2lino(struct gnode *ga, /* gnode address */
```
Comment (BenBE) [inline lib/dialects/aix/dnode2.c:55]: Missing `ctx` argument

File: lib/dialects/aix/dproc.c [hunk 10]
```diff
@@ -256,12 +256,12 @@ char *ea;                         /* expected architecture */
 
         (void)fprintf(stderr, "%s: can't execute %s: %s\n", Pn, OSLEVELPATH,
                       strerror(errno));
-        Error();
+        Error(ctx);
```
Comment (BenBE) [inline lib/dialects/aix/dproc.c:259]: Missing `ctx` argument

File: lib/dialects/aix/dproc.c [hunk 11]
```diff
@@ -778,7 +778,7 @@ static void get_kernel_access() {
                       kbb);
         (void)fprintf(stderr, "      The bit size of this kernel is %s.\n",
                       kbr);
-        Error();
+        Error(ctx);
```
Comment (BenBE) [inline lib/dialects/aix/dproc.c:781]: Missing `ctx` argument

**Review by BenBE — CHANGES_REQUESTED** [review]
Still a lot of functions missing the `ctx` argument …
Please report back, once you have completed that cleanup.

**Comment by jiegec** [comment]
> Still a lot of functions missing the `ctx` argument … Please report back, once you have completed that cleanup.

Thanks, I don't have AIX, HP-UX or UnixWare environment, It is hard to enumerate those places without compilers help.

**Comment by jiegec** [comment]
I installed SCO OpenServer 5 in VM, but it is too hard to use from 2023's perspective.. I installed git, make etc. but was unable to build lsof. I surrender.

**Comment by BenBE** [comment]
Looking through the diff, there are (still) several (easy to spot) places, where the `ctx` is used, but not provided as an argument. This will likely give a warning, but given that there's a global variable in the main process somewhat compile nonetheless. This is extremely unclean code style, and does not usually require to compile the code (TBF: I did just have a look at the code and spotted those issues). I know hpux, aix and several other platforms are rare to come by for testing, but a minimum of diligence should be used when working on such patch sets.

**Comment by jiegec** [comment]
The global variable in the main process is removed.

**Comment by BenBE** [comment]
Thanks for this note.

Seems as if GH was showing me some outdated (partial) version of the diff earlier for this patch; looks much cleaner now.

**Pull request merged**